File: Samples/Pages and Navigation/HeaderAndFooter/C#/HeaderAndFooter.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class HeaderAndFooter
    {
        public static void Main()
        {
            // NOTE:
            // When used in trial mode, the library imposes some restrictions.
            // Please visit http://bitmiracle.com/pdf-library/trial-restrictions.aspx
            // for more information.
            
            const string PathToFile = "HeaderAndFooter.pdf";

            using (var pdf = new PdfDocument(@"..\Sample Data\BRAILLE CODES WITH TRANSLATION.pdf"))
            {
                PdfFont font = pdf.AddFont(PdfBuiltInFont.Helvetica);

                for (int i = 0; i < pdf.PageCount; ++i)
                {
                    PdfPage page = pdf.Pages[i];
                    DrawHeader(page, font);
                    DrawFooter(i, page, font);
                }

                pdf.Save(PathToFile);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(PathToFile) { UseShellExecute = true });
        }

        private static void DrawHeader(PdfPage page, PdfFont font)
        {
            const string HeaderText = "Header";

            var canvas = page.Canvas;
            canvas.Font = font;
            canvas.FontSize = 30;

            double textWidth = canvas.GetTextWidth(HeaderText);
            double rotatedPageWidth = GetRotatedPageSize(page).Width;
            var headerPosition = new PdfPoint((rotatedPageWidth - textWidth) / 2, 10);

            ShowTextAtRotatedPage(HeaderText, headerPosition, page);
        }

        private static void DrawFooter(int pageIndex, PdfPage page, PdfFont font)
        {
            var canvas = page.Canvas;
            canvas.Font = font;
            canvas.FontSize = 14;

            PdfSize rotatedPageSize = GetRotatedPageSize(page);
            var paddingFromCorner = new PdfSize(30, 30);
            var positionRightBottom = new PdfPoint(
                rotatedPageSize.Width - paddingFromCorner.Width,
                rotatedPageSize.Height - paddingFromCorner.Height
            );
            
            ShowTextAtRotatedPage((pageIndex + 1).ToString(), positionRightBottom, page);
        }

        private static PdfSize GetRotatedPageSize(PdfPage page)
        {
            if (page.Rotation == PdfRotation.Rotate90 || page.Rotation == PdfRotation.Rotate270)
                return new PdfSize(page.Height, page.Width);

            return new PdfSize(page.Width, page.Height);
        }

        private static void ShowTextAtRotatedPage(string text, PdfPoint position, PdfPage page)
        {
            PdfCanvas canvas = page.Canvas;
            if (page.Rotation == PdfRotation.None)
            {
                canvas.DrawString(position.X, position.Y, text);
                return;
            }

            canvas.SaveState();

            switch (page.Rotation)
            {
                case PdfRotation.Rotate90:
                    canvas.TranslateTransform(position.Y, page.Height - position.X);
                    break;

                case PdfRotation.Rotate180:
                    canvas.TranslateTransform(page.Width - position.X, page.Height - position.Y);
                    break;

                case PdfRotation.Rotate270:
                    canvas.TranslateTransform(page.Width - position.Y, position.X);
                    break;
            }
            canvas.RotateTransform((int)page.Rotation * 90);
            canvas.DrawString(0, 0, text);

            canvas.RestoreState();
        }
    }
}

File: Samples/Pages and Navigation/HeaderAndFooter/VB.NET/HeaderAndFooter.vb
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class HeaderAndFooter
        Public Shared Sub Main()
            ' NOTE:
            ' When used in trial mode, the library imposes some restrictions.
            ' Please visit https://bitmiracle.com/pdf-library/trial-restrictions
            ' for more information.

            Const PathToFile As String = "HeaderAndFooter.pdf"

            Using pdf = New PdfDocument("..\Sample Data\BRAILLE CODES WITH TRANSLATION.pdf")
                Dim font As PdfFont = pdf.AddFont(PdfBuiltInFont.Helvetica)

                For i As Integer = 0 To pdf.PageCount - 1
                    Dim page As PdfPage = pdf.Pages(i)
                    DrawHeader(page, font)
                    DrawFooter(i, page, font)
                Next

                pdf.Save(PathToFile)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(PathToFile) With {.UseShellExecute = True})
        End Sub

        Private Shared Sub DrawHeader(page As PdfPage, font As PdfFont)
            Dim canvas = page.Canvas
            canvas.Font = font
            canvas.FontSize = 30

            Const HeaderText As String = "Header"
            Dim textWidth As Double = canvas.GetTextWidth(HeaderText)
            Dim rotatedPageWidth As Double = GetRotatedPageSize(page).Width
            Dim headerPosition = New PdfPoint((rotatedPageWidth - textWidth) / 2, 10)

            ShowTextAtRotatedPage(HeaderText, headerPosition, page)
        End Sub

        Private Shared Sub DrawFooter(pageIndex As Integer, page As PdfPage, font As PdfFont)
            Dim canvas = page.Canvas
            canvas.Font = font
            canvas.FontSize = 14

            Dim rotatedPageSize As PdfSize = GetRotatedPageSize(page)
            Dim paddingFromCorner = New PdfSize(30, 30)
            Dim positionRightBottom = New PdfPoint(rotatedPageSize.Width - paddingFromCorner.Width, rotatedPageSize.Height - paddingFromCorner.Height)

            ShowTextAtRotatedPage((pageIndex + 1).ToString(), positionRightBottom, page)
        End Sub

        Private Shared Function GetRotatedPageSize(page As PdfPage) As PdfSize
            If page.Rotation = PdfRotation.Rotate90 OrElse page.Rotation = PdfRotation.Rotate270 Then
                Return New PdfSize(page.Height, page.Width)
            End If

            Return New PdfSize(page.Width, page.Height)
        End Function

        Private Shared Sub ShowTextAtRotatedPage(text As String, position As PdfPoint, page As PdfPage)
            Dim canvas As PdfCanvas = page.Canvas

            If page.Rotation = PdfRotation.None Then
                canvas.DrawString(position.X, position.Y, text)
                Return
            End If

            canvas.SaveState()

            Select Case page.Rotation
                Case PdfRotation.Rotate90
                    canvas.TranslateTransform(position.Y, page.Height - position.X)
                Case PdfRotation.Rotate180
                    canvas.TranslateTransform(page.Width - position.X, page.Height - position.Y)
                Case PdfRotation.Rotate270
                    canvas.TranslateTransform(page.Width - position.Y, position.X)
            End Select

            canvas.RotateTransform(CInt(page.Rotation) * 90)
            canvas.DrawString(0, 0, text)

            canvas.RestoreState()
        End Sub
    End Class
End Namespace